src/rules/load-partial-extension/index.js

```diff
@@ -0,0 +1,143 @@
+"use strict";
+
+const nodeJsPath = require("path");
+const { utils } = require("stylelint");
+const atRuleParamIndex = require("../../utils/atRuleParamIndex");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("load-partial-extension");
+
+const messages = utils.ruleMessages(ruleName, {
+  expected: (rule = "import") => `Expected @${rule} to have an extension`,
+  rejected: (ext, rule = "import") =>
+    `Unexpected extension ".${ext}" in @${rule}`
+});
+
+const meta = {
+  url: ruleUrl(ruleName)
+};
+
+// https://drafts.csswg.org/mediaqueries/#media-types
+const mediaQueryTypes = [
+  "all",
+  "print",
+  "screen",
+  "speech",
+  "tv",
+  "tty",
+  "projection",
+  "handheld",
+  "braille",
+  "embossed",
+  "aural"
+];
+
+const mediaQueryTypesRE = new RegExp(`(${mediaQueryTypes.join("|")})$`, "i");
+const hasArgumentsRegExp = /\(\s*([^)]+?)\s*\)/;
+const stripPath = path =>
+  path.replace(/^\s*(["'])\s*/, "").replace(/\s*(["'])\s*$/, "");
+
+function rule(expectation, _, context) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: ["always", "never"]
+    });
+    let isLoadCss = false;
+
+    if (!validOptions) {
+      return;
+    }
+
+    root.walkAtRules(atRule => {
+      if (
+        atRule.name === "import" ||
+        atRule.name === "use" ||
+        atRule.name === "forward" ||
+        atRule.name === "include"
```

Comment:
Small detail, but the code could be a bit easier to read by doing something like this:

```js
const loadAtRules = ["import", "use", "forward", "include"];

// https://drafts.csswg.org/mediaqueries/#media-types
const mediaQueryTypes = [
```

---

```js
    root.walkAtRules(atRule => {
      if (loadAtRules.includes(atRule.name)) {

```